app/assets/javascripts/discourse/app/components/composer/toolbar-buttons.gjs

```diff
@@ -0,0 +1,68 @@
+import Component from "@glimmer/component";
+import { fn, hash } from "@ember/helper";
+import { action } from "@ember/object";
+import { eq } from "truth-helpers";
+import DButton from "discourse/components/d-button";
+import concatClass from "discourse/helpers/concat-class";
+import icon from "discourse/helpers/d-icon";
+import ToolbarPopupMenuOptions from "select-kit/components/toolbar-popup-menu-options";
+
+export default class ComposerToolbarButtons extends Component {
+  firstButton = this.args.data.groups.mapBy("buttons").flat().firstObject;
+
+  @action
+  applyCondition(button) {
+    return button.condition === undefined
+      ? true
+      : button.condition(this.args.data);
+  }
+
+  @action
+  tabIndex(button) {
+    return button === this.firstButton ? 0 : button.tabindex;
+  }
+
+  <template>
+    {{#each @data.groups key="group" as |group|}}
+      {{#each group.buttons key="id" as |button|}}
+        {{#if (this.applyCondition button)}}
+          {{#if button.href}}
+            <a
+              href={{button.href}}
+              target="_blank"
+              rel="noopener noreferrer"
+              class={{concatClass "btn no-text btn-icon" button.className}}
+              title={{button.title}}
+              tabindex={{this.tabIndex button}}
+            >
+              {{icon button.icon}}
```

Comment:
Is it possible for the definition to also have a label specified here?

Reply:
Nothing of this is new, I'm just moving existing code around...  I'd defer adding a label for another time if possible, which an example at hand